mono/mini/method-to-ir.c

```diff
@@ -6113,6 +6120,8 @@ inline_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsig,
 	cfg->disable_inline = prev_disable_inline;
 	cfg->inline_depth --;
 
+	clear_cfg_error (cfg);
```

Comment:
This is alright, but maybe we could push the error handling down into the two branches of the `if`, below, and try to get something out of the error:

1. If we decide to inline, can't we check that `cfg->error` must have been ok?
   ```c
 		if (cfg->verbose_level > 2)
  			printf ("INLINE END %s -> %s\n", mono_method_full_name (cfg->method, TRUE), mono_method_full_name (cmethod, TRUE));
   ```
2. If we find out we can't inline, can't we check if the error is not ok and print it out if the verbosity is high enough?
   ```c
  		if (cfg->verbose_level > 2)
 			printf ("INLINE ABORTED %s (cost %d)\n", mono_method_full_name (cmethod, TRUE), costs);
   ```

Reply:
If the error is set method_to_ir () returns -1 i.e. an error, so the inline path won't happen.

Reply:
Could it happen that `inline_always` is true but `costs == -1` and `cfg->error` is set?